src/main/java/net/minecraftforge/common/world/IDimensionExtra.java

```diff
@@ -0,0 +1,33 @@
+package net.minecraftforge.common.world;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public interface IDimensionExtra
+{
+    Codec<DimensionExtraType<?>> FROM_REGISTRY = ResourceLocation.field_240908_a_.xmap(ForgeRegistries.DIMENSION_EXTRA_TYPE::getValue, ForgeRegistries.DIMENSION_EXTRA_TYPE::getKey);
+    Codec<IDimensionExtra> GENERAL_CODEC = FROM_REGISTRY.dispatch(IDimensionExtra::getType, DimensionExtraType::getCodec);
+
+    DimensionExtraType<?> getType();
+
+    /**
+     * Errors if the list contains 2 objects with the same "type" key.
+     */
+    static DataResult<List<IDimensionExtra>> ensureUnique(List<IDimensionExtra> extras)
+    {
+        Set<ResourceLocation> toTest = new HashSet<>();
+        for(IDimensionExtra extra : extras)
+        {
+            if(!toTest.add(extra.getType().getRegistryName()))
+                return DataResult.error("Duplicate founds in forge extras!", extras);
```

Comment:
```suggestion
                return DataResult.error("Duplicate dimension type extra entries were found: " + extra.getType().getRegistryName(), extras);
```
Mojang's DFU errors are already horrible to understand, providing a little more information would be super nice here. Even better if the dimension type registry name could be appended to this string to identify *where* the extras were found.

Reply:
well the dimension type is being created when this called. so no real way to know what its registry name is.